### PR TITLE
0.5.0-beta

### PR DESCRIPTION
**Pull request by maZahaca**
Features: #59 #60 #57 #66 #64

### CODE REVIEW

File: lib/Parser.js [hunk 1]
```diff
@@ -397,6 +397,11 @@ Parser.prototype = {
             if (rule.set) {
                 this._actions.set(rule.set, results);
             }
+            if (rule.add) {
+                const cuttent = this._actions.get(rule.add) || [];
```
Comment (jifeon) [inline lib/Parser.js]: `current` you mean?

Comment (maZahaca) [inline lib/Parser.js]: sure 👊

**Comment by jifeon** [comment]
alpha 42 reviewed, nice version lol ))) so wanna get beta

File: package.json [hunk 2]
```diff
@@ -60,7 +60,8 @@
     "phantom": "^0.8.4",
     "phantomjs-prebuilt": "^2.1.7",
     "vow": "^0.4.10",
-    "vow-node": "^0.3.0"
+    "vow-node": "^0.3.0",
+    "html-entities": "^1.2.0"
```
Comment (jifeon) [inline package.json]: what does it do?

Comment (jifeon) [inline package.json]: and why do we need it? ))

Comment (maZahaca) [inline package.json]: It allows to do one more transformation which is required for several parsers - decodeHTML.
We need to think about ability to specify what actions and transformation you need to build the best match package of goose.

File: lib/Parser.js [hunk 3]
```diff
@@ -290,15 +289,23 @@ Parser.prototype = {
...
      */
-    addAction: function() {
-        return actionsFactory.addAction.apply(actionsFactory, arguments)
+    addAction: function(type, action) {
+        return this._actions.addAction(type, action);
     },
 
     /**
-     * @see {@link Transformations#addTransformation}
+     * @deprecated use addTransform instead
```
Comment (jifeon) [inline lib/Parser.js]: Goose. Made with care about users ))

File: lib/Actions.js [hunk 4]
```diff
@@ -147,76 +149,94 @@ Actions.prototype = {
...
 
     /**
+     * @param action
+     * @param selector
+     * @param parentSelector
+     * @param prevResult
+     * @return {Action}
+     * @private
+     */
+    _createInstance(action, selector, parentSelector, prevResult) {
```
Comment (jifeon) [inline lib/Actions.js]: `_createInstance` is ambiguous name, maybe `_createAction` or `_createActionInstance` ?

File: tests/phantom/lib/actions/actionsSpec.js [hunk 5]
```diff
@@ -8,6 +8,168 @@ describe('Using actions, the parser', () => {
         parser = createParser();
     });
 
+    it('condition.then', async () => {
```
Comment (jifeon) [inline tests/phantom/lib/actions/actionsSpec.js]: It's specs. Name of the suite + name of the test should form and a meaningful sentence.

File: tests/phantom/lib/actions/actionsSpec.js [hunk 6]
```diff
@@ -8,6 +8,168 @@ describe('Using actions, the parser', () => {
...
+    });
+
+    it('cases.2', async () => {
+        setServerResponse(
+            `
+                <a class='query' href="tel:834775566654">query</a>
+                <a class='page' href="/page?phone=834775566654">page</a>
+            `,
+            function run () {
+                document.querySelector('.page').innerText = window.location.search.slice(7);
```
Comment (jifeon) [inline tests/phantom/lib/actions/actionsSpec.js]: why you need it here?

File: tests/phantom/lib/actions/actionsSpec.js [hunk 7]
```diff
@@ -8,6 +8,168 @@ describe('Using actions, the parser', () => {
...
+                            }
+                        ]
+                    ]
+                }
+            ],
+            rules: {
+                scope: '.page'
+            }
+        });
+        console.log(result);
```
Comment (jifeon) [inline tests/phantom/lib/actions/actionsSpec.js]: remove `console.log`s before commit

**Comment by jifeon** [comment]
That's what I call great branch removing!

**Pull request merged**